File: crates/vexus-index/tests/fixtures/swift/sample.swift
import Foundation

struct Point {
    var x: Int
    var y: Int
}

// One shopping cart.
class Cart {
    var items: [Int] = []

    func total(tax: Int) -> Int {
        return subtotal() + tax
    }

    private func subtotal() -> Int {
        return items.reduce(0, +)
    }
}

enum Color {
    case red
    case green
}

protocol Renderer {
    func render() -> String
}

extension Cart {
    func describe() -> String {
        return "cart with \(items.count)"
    }
}

func fetchTotal(cart: Cart) -> Int {
    return cart.total(tax: 5)
}

File: crates/vexus-index/tests/fixtures/dart/sample.dart
import 'package:shop/pricing.dart';

class Cart {
  final List<int> items;

  Cart(this.items);

  // Total in cents.
  int total(int tax) {
    return subtotal() + tax;
  }

  int subtotal() => items.fold(0, (a, b) => a + b);
}

mixin Loggable {
  void log(String msg) {}
}

enum Color { red, green }

extension CartX on Cart {
  String describe() => 'cart';
}

int checkoutTotal(Cart cart, int tax) {
  return cart.total(tax);
}
